metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -168,18 +169,173 @@ void main() {
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        final currentConfiguration = notifier.currentConfiguration;
+
+        expect(currentConfiguration.name, equals(RouteName.login));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not clear pages when a user is not logged in",
+      () {
+        notifier.push(DefaultRoutes.login);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(2));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not push to the login page when a user is not logged in",
+      () {
+        final initialPagesLength = notifier.pages.length;
+
+        notifier.handleLoggedOut();
+
+        final actualPagesLength = notifier.pages.length;
+
+        expect(actualPagesLength, equals(initialPagesLength));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the dashboard route if the current redirect route is null",
+      () {
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.dashboard));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears pages before redirect",
+      () {
+        notifier.handleAppInitialized(isAppInitialized: true);
+        notifier.push(DefaultRoutes.projectGroups);
+
+        notifier.handleLoggedIn();
+
+        expect(notifier.pages, hasLength(equals(1)));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the redirect route if it is not null",
+      () {
+        final configuration = DefaultRoutes.projectGroups;
+
+        notifier.push(configuration);
+
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(configuration.name));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears the redirect route after redirect if a user is logged in",
+      () {
+        notifier.push(DefaultRoutes.projectGroups);
+
+        notifier.handleLoggedIn();
+
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.dashboard));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() does not clear pages when a user is already logged in",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.projectGroups);
+
+        final initialPagesLength = notifier.pages.length;
+
+        notifier.handleLoggedIn();
+
+        final actualPagesLength = notifier.pages.length;
+
+        expect(actualPagesLength, equals(initialPagesLength));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() does not push to the redirect route when a user is already logged in",
```

Comment:
✅ 
```suggestion
      ".handleLoggedIn() does not push if a user is already logged in",
```
I believe we should test that it just does not push anywhere if the user is already logged in.